CT-1869 Fix bug caused by deleting linked cases

stephenrichards opened the pull request:
If a case had linked cases, and one of the linked cases was subsequently
soft deleted, that case became invalid, and could no longer be updated due
to failing validation errors.

This is fixed by the removal of the  `vaidates :linked_case` on the 'LinkedCase'
model.  This is not required, as there are validations for either linked_case id or
linked case number on creation.

misaka commented on app/models/business_unit.rb:
```diff
@@ -49,11 +49,19 @@ def valid_role
+    code == Settings.foi_cases.default_clearance_team
+  end
+
+  def self.dacu_bmt
```
Let's make sure we have tests for this method and `dacu_bmt?` below.

misaka commented on db/structure.sql:
```diff
@@ -2,11 +2,12 @@
--- Dumped from database version 9.5.1
--- Dumped by pg_dump version 9.5.1
+-- Dumped from database version 9.6.5
+-- Dumped by pg_dump version 9.6.5
```
As @aliuk2012 has mentioned, I think, we need to make sure we're all on the same version of PostgreSQL, which is dictated by our deployed environments in AWS: 9.5.

stephenrichards committed:
CT-1869 Fix bug caused by deleting linked cases

If a case had linked cases, and one of the linked cases was subsequently
soft deleted, that case became invalid, and could no longer be updated due
to failing validation errors.

This is fixed by the removal of the  `vaidates :linked_case` on the 'LinkedCase'
model.  This is not required, as there are validations for either linked_case id or
linked case number on creation.
